cljfmt/src/cljfmt/main.clj

```diff
@@ -154,65 +154,86 @@
       (read-string contents)
       (edn/read-string contents))))
 
-(def default-options
+(defn- parent-dirs [^String root]
+  (->> (.getAbsoluteFile (io/file root))
+       (iterate #(.getParentFile ^java.io.File %))
+       (take-while some?)))
+
+(defn- find-file-in-dir ^java.io.File [^java.io.File dir ^String name]
+  (let [f (io/file dir name)]
+    (when (.exists f) f)))
+
+(defn- find-config-file-in-dir ^java.io.File [^java.io.File dir]
+  (or (find-file-in-dir dir ".cljfmt.edn")
+      (find-file-in-dir dir ".cljfmt.clj")))
+
+(defn- load-configuration []
+  (some->> (parent-dirs "")
+           (some find-config-file-in-dir)
+           (#(.getPath ^java.io.File %))
```

Comment:
Couldn't this be `str`? Or we add a `str` to `filename-ext` in order to make that safe for files.

Reply:
`str` works, but without the type hinting I get `java.io.File cannot be cast to java.lang.String` at runtime with GraalVM, so it'd look similar here. There could be an explicit type check in `filename-ext` as well, I guess.